src/utils/db/db.ts

```diff
@@ -496,6 +496,49 @@ export class DB {
         return person
     }
 
+    public async setPersonAsIdentified(person: Person): Promise<void> {
+        let shouldUpdateClickhouse = false
+
+        const result = await this.postgresQuery(
+            `
+                UPDATE posthog_person 
+                SET is_identified=$1 
+                WHERE id = $2 
+                AND is_identified = $3
+                RETURNING is_identified
+            `,
+            [true, person.id, false],
+            'setPersonAsIdentified'
+        )
+
+        if (result.rows.length > 0) {
+            shouldUpdateClickhouse = true
+        }
+
+        if (this.kafkaProducer && shouldUpdateClickhouse) {
```

Comment:
What if we crashed just before this line last time, how can I now setIdentified for the clickhouse person (the postgres person is already identified, so calling twice won't help) && in general what happens if we have people identified in Postgres but not in CH?

Reply:
I have an idea

Reply:
Wrapped it in a transaction, so that if we don't queue the message in Kafka, we'll rollback on Postgres too, making sure we keep both in the same state and a subsequent identify call will get them both back where they should be.